Add my GitHub action for setting up the CLI

ajhalili2006 opened the pull request:
I'm currently working on this GitHub action to easily install the Railway CLI through npm global installs and the official install script. Currently, I'm working on fixing issues on building from the source (especially if you want to try new CLI features before Deploy Fridays) and Windows stuff.

Commit:
Add my GitHub action for setting up the CLI.

FarazPatankar commented on README.md:
```diff
@@ -17,6 +17,7 @@ A curated list of awesome resources related to [Railway](https://railway.app/).
   - [Utilities](#utilities)
 - [Community](#community)
   - [Built on Railway](#built-on-railway)
+  - [GitHub Actions](#github-actions)
```
This is a little too specific IMO. Can we change this to Utilities and make it more generic?

FarazPatankar commented on README.md:
```diff
@@ -63,7 +64,12 @@ A list of starters built by the community using the [Railway button](https://rai
+### GitHub Actions
+A list of community-maintained GitHub Actions for Railway.
   
+- [Setup Railway CLI Action](https://github.com/MadeByThePinsHub/setup-railway-cli-action) -  An GitHub action to set up Railway CLI through either npm global install and install script for Linux and macOS. Additional features like version locking and building from source will be available in the future.
```
Small change, this should be "A GitHub action".

FarazPatankar reviewed (changes requested):
Thanks for the PR, @AndreiJirohHaliliDev2006. Left a couple of comments.

Commit:
chore(readme): update readme based on feedback

Source: https://github.com/railwayapp/awesome-railway/pull/4#pullrequestreview-688406227

Signed-off-by: Andrei Jiroh Eugenio Halili <andreijiroh@madebythepins.tk>
